lib/inventory/controller_test.go

```diff
@@ -223,6 +268,239 @@ func TestControllerBasics(t *testing.T) {
 	require.Zero(t, unexpectedAddrs)
 }
 
+// TestInstanceHeartbeat verifies basic expected behaviors for instance heartbeat.
+func TestInstanceHeartbeat(t *testing.T) {
```

Comment:
This tripped the race detector on the first attempt:
```
go test ./lib/inventory/ -run TestInstanceHeartbeat -race
==================
WARNING: DATA RACE
Read at 0x00c0000c4a18 by goroutine 20:
  github.com/gravitational/teleport/api/types.(*InstanceV1).GetControlLog()
      /Users/zmb/teleport/api/types/instance.go:215 +0x30
  github.com/gravitational/teleport/lib/inventory.(*InstanceStateTracker).nextHeartbeat()
      /Users/zmb/teleport/lib/inventory/inventory.go:374 +0x1ec
  github.com/gravitational/teleport/lib/inventory.(*Controller).heartbeatInstanceState()
      /Users/zmb/teleport/lib/inventory/controller.go:335 +0x30c
  github.com/gravitational/teleport/lib/inventory.(*Controller).handleControlStream()
      /Users/zmb/teleport/lib/inventory/controller.go:260 +0x354
  github.com/gravitational/teleport/lib/inventory.(*Controller).RegisterControlStream.func1()
      /Users/zmb/teleport/lib/inventory/controller.go:188 +0x40

Previous write at 0x00c0000c4a18 by goroutine 8:
  github.com/gravitational/teleport/api/types.(*InstanceV1).AppendControlLog()
      /Users/zmb/teleport/api/types/instance.go:219 +0xd8
  github.com/gravitational/teleport/lib/inventory.TestInstanceHeartbeat()
      /Users/zmb/teleport/lib/inventory/controller_test.go:310 +0x540
  testing.tRunner()
      /opt/homebrew/Cellar/go/1.19.3/libexec/src/testing/testing.go:1446 +0x188
  testing.(*T).Run.func1()
      /opt/homebrew/Cellar/go/1.19.3/libexec/src/testing/testing.go:1493 +0x40
```

Reply:
Ah, forgot to add a deep copy to `fakeAuth.CompareAndSwapInstance`.